File: day88/cf.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin >> t;
     while (t--){
        string s; cin>>s;
        // int n = s.size();
        bool flag=true;
        int i=0;
        // cout<<s.substr(i,i+2)<<endl;
        // cout<<i+3<<endl;
        while(i<s.size()){
            if(s.substr(i,i+3)=="aaa"){
                i=i+3;
                cout << i<<"-->aaa" << endl;
            }
            else if(s.substr(i,i+2)=="aa"){
                i=i+2;
                cout << i << "-->aa" << endl;
            }
            else if(s.substr(i,i+3)=="bbb"){
                i=i+3;
                cout << i << "-->bbb" << endl;
            }
            else if(s.substr(i,i+2)=="bb"){                
                i=i+2;
                cout << i << "-->bb" << endl;
            }
            else{
                flag=false;
                break;
            }
        }
        if(flag){
            cout<<"YES"<<endl;
        }
        else{
            cout<<"NO"<<endl;
        }
     }
    return 0;
}

File: day88/cf.py
#string building

# You are given a string s. You have to determine whether it is possible to build the string s out of strings aa, aaa, bb and/or bbb by concatenating them. You can use the strings aa, aaa, bb and/or bbb any number of times and in any order.

def checkio(string):
    flag=True
    for i in range(len(string)):
        
        if string[i] == 'a' and i+2 or i+3<len(string):
            if string[i:i+2] == 'aa' or string[i:i+3] == 'aaa':
                print(string[i:i+2], string[i:i+3])
                flag=True
                continue
            else:
                flag=False
        elif string[i] == 'b' and i+2 or i+3<len(string):
            if string[i:i+2] == 'bb' or string[i:i+3] == 'bbb':
                print(string[i:i+2], string[i:i+3])
                flag=True
                continue
            else:
                flag=False

    return flag

string = input()
print(checkio(string))

